resources/type-system/flow-analysis.md

```diff
@@ -433,7 +468,7 @@ Definitions:
       - Else if `S` is `X extends R` then let `T1` = `X & T`
       - Else If `S` is `X & R` then let `T1` = `X & T`
       - Else `x` is not promotable (shouldn't happen since we checked above)
-      - Let `VM2 = VariableModel(declared, T1::promoted, T1::tested, assigned,
+      - Let `VM2 = VariableModel(declared, T1::promoted, S::tested, assigned,
```

Comment:
Is this a deliberate change?  It seems wrong--we want to add `T1` to the "tested" set, since it's (effectively) the type we're testing against.  `S` is the *old* type, so adding it to "tested" doesn't gain us anything.

Reply:
Yes, `S` was wrong.  It's not clear to me that `T1` is right though, is it?  Or should it be `T`?

Reply:
Good point.  The current implementation uses `T`, so let's change it to that.  I'm happy to do that as a separate PR if you want (since at this point it's unrelated to this change), or you can go ahead and do it now--your call.

Reply:
Since I backed out the rest of the changes to this doc, how about you add this change to one of your updates?

Reply:
Ok, will do.

Reply:
https://github.com/dart-lang/language/pull/1116